packages/db/tests/translate/test-granules.js

```diff
@@ -1,7 +1,431 @@
+  });
+
+  await t.throwsAsync(translatePostgresGranuleToApiGranule({
+    granulePgRecord: postgresGranule,
+    collectionPgRecord: collection,
+    knexOrTransaction: knex,
+    collectionPgModel,
+    pdrPgModel,
+    providerPgModel,
+    filePgModel,
+  }),
+  { instanceOf: ValidationError });
+});
+
+test('translatePostgresGranuleToApiGranule does not require a PDR or Provider', async (t) => {
+  const {
+    knex,
+    pdrPgModel,
+    providerPgModel,
+    collectionPgModel,
+    filePgModel,
+    granulePgModel,
+    granuleCumulusId,
+    executions,
+  } = t.context;
+
+  const postgresGranule = await granulePgModel.get(knex, { cumulus_id: granuleCumulusId });
+
+  delete postgresGranule.pdr_cumulus_id;
+  delete postgresGranule.provider_cumulus_id;
+
+  const expectedApiGranule = {
+    beginningDateTime: postgresGranule.beginning_date_time.toISOString(),
+    cmrLink: postgresGranule.cmr_link,
+    collectionId: 'collectionName___collectionVersion',
+    createdAt: postgresGranule.created_at.getTime(),
+    duration: postgresGranule.duration,
+    endingDateTime: postgresGranule.ending_date_time.toISOString(),
+    error: postgresGranule.error,
+    execution: getExecutionUrlFromArn(executions[0].arn),
+    granuleId: postgresGranule.granule_id,
+    lastUpdateDateTime: postgresGranule.last_update_date_time.toISOString(),
+    processingEndDateTime: postgresGranule.processing_end_date_time.toISOString(),
+    processingStartDateTime: postgresGranule.processing_start_date_time.toISOString(),
+    productionDateTime: postgresGranule.production_date_time.toISOString(),
+    productVolume: postgresGranule.product_volume,
+    published: postgresGranule.published,
+    queryFields: postgresGranule.query_fields,
+    status: postgresGranule.status,
+    timestamp: postgresGranule.timestamp.getTime(),
+    timeToArchive: postgresGranule.time_to_archive,
+    timeToPreprocess: postgresGranule.time_to_process,
+    updatedAt: postgresGranule.updated_at.getTime(),
+    files: [
+      {
+        bucket: 'cumulus-test-sandbox-private',
+        checksum: 'bogus-value',
+        checksumType: 'md5',
+        fileName: 's3://cumulus-test-sandbox-private/firstKey',
+        key: 'firstKey',
+        size: '2098711627776',
+        source: 's3://cumulus-test-sandbox-private/sourceDir/granule',
+
+      },
+      {
+        bucket: 'cumulus-test-sandbox-private',
+        checksum: 'bogus-value',
+        checksumType: 'md5',
+        fileName: 's3://cumulus-test-sandbox-private/secondKey',
+        key: 'secondKey',
+        size: '1099511627776',
+        source: 's3://cumulus-test-sandbox-private/sourceDir/granule',
+      },
+    ],
+  };
+
+  const result = await translatePostgresGranuleToApiGranule({
+    granulePgRecord: postgresGranule,
+    knexOrTransaction: knex,
+    collectionPgModel,
+    pdrPgModel,
+    providerPgModel,
+    filePgModel,
+  });
+
+  t.deepEqual(
+    result,
+    expectedApiGranule
+  );
+});
```

Comment:
These tests were pulled from Rds-Phase2 as part of the move of translatePostgresGranuleToApiGranule()